napari/_qt/qt_viewer.py

```diff
@@ -497,6 +503,9 @@ def console(self):
                     self.console.push(
                         {'napari': napari, 'action_manager': action_manager}
                     )
+                    for i in self.console_backlog:
+                        self.console.push(i)
+                    self._console_backlog = []
```

Comment:
@Czaki and I discussed this idea at the euro/asia community meeting yesterday. @Czaki's concern (if I'm not butchering it) is that this list could lead to memory "leaks" if a plugin adds things and then the plugin is closed down without the console ever getting called. His suggestion (again I might be butchering it) is to use *weakrefs* when adding to the backlog, and then here you convert them to real refs. Does that make sense?

Reply:
Yes, makes sense, I was wondering about memory issues, too, but didn't know a solution. I'll look into this option.

Reply:
I added in logic to make weakrefs when possible when adding to backlog. Let me know if things look OK.